.github/workflows/release-checklist.yml

```
@@ -12,51 +10,64 @@ concurrency:
   cancel-in-progress: true
 
 permissions:
-  contents: read
+  contents: write
   issues: write
+  pull-requests: write
 
 jobs:
-  release-checlist:
+  release-check:
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
 
       - name: Set up Python
         uses: actions/setup-python@v6
         with:
           python-version: "3.x"
 
-      - name: Install deps
+      - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install PyGithub
+          pip install PyGithub PyYAML
```

```suggestion
          python -m pip install PyGithub PyYAML
```
though the cleaner way would be to do this via a `requirements.txt` file 
```suggestion
          python -m pip install -r etc/requirements.txt
```
with `requirements.txt` having this content:
```
PyGitHub
PyYAML
```
so that anyone who wants to use this code interactively (outside of the actions) can do use this as well.

Alas this is less important overall, :-)